framework/src/base/MooseApp.C

```diff
@@ -562,20 +600,45 @@ MooseApp::runInputFile()
 
   _action_warehouse.executeAllActions();
 
-  if (isParamValid("mesh_only"))
+  if (getParam<std::string>("split_mesh") != "")
   {
-    writeMeshOnly(getParam<std::string>("mesh_only"));
     _ready_to_exit = true;
+    auto & mesh = _action_warehouse.mesh();
+
+    if (mesh->getFileName() == "" && getParam<std::string>("split_file") == "")
+      mooseError("must specify an output mesh file name (--split-file) when splitting "
+                 "non-file-based meshes");
+
+    std::vector<unsigned int> splits = splitstr(getParam<std::string>("split_mesh"));
+    for (std::size_t i = 0; i < splits.size(); i++)
+    {
+      processor_id_type n = splits[i];
+      Moose::out << "splitting " << n << " ways..." << std::endl;
+
+      auto cpr = libMesh::split_mesh(*mesh, n);
```

Comment:
Does the reset to 1 partition happen inside `split_mesh()`?

Reply:
Also: maybe the libMesh interface should really just take the vector of splits... and do this loop internally.

Reply:
Yes.  I thought about that, but this allows the callers of the API more flexibility on how they want to handle the checkpoint object - they can switch it into ascii mode, etc. without having to pass a bajillion options into the splitting API.  The goal was to make the libmesh splitting API do as close to one thing as possible and still preserve flexibility for callers.